src/components/TodoItem.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useRef, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { deleteTodoFromServer, updateTodoOnServer } from '../api/todos';
+import { TodoError } from '../types/TodoError';
+
+type Props = {
+  todo: Todo,
+  deleteTodo: (todoId: number) => void,
+  setErrorMessage: (newError: TodoError) => void,
+  updateTodo: (updatedTodo: Todo) => void,
+  loadingTodos: number[],
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  deleteTodo,
+  setErrorMessage,
+  updateTodo,
+  loadingTodos,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [isEditing, setIsEditing] = useState(false);
+  const [title, setTitle] = useState(todo.title);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const setInputFocus = () => {
+    setTimeout(() => {
+      if (inputRef.current) {
+        inputRef.current.focus();
+      }
+    }, 0);
+  };
+
+  const handleRemoveClick = () => {
+    setIsLoading(true);
+
+    deleteTodoFromServer(todo.id)
+      .then(() => deleteTodo(todo.id))
+      .catch(() => setErrorMessage(TodoError.Delete))
+      .finally(() => setIsLoading(false));
+  };
+
+  const handleCheckboxChange = () => {
+    setIsLoading(true);
+
+    const updatedTodo: Todo = {
+      ...todo,
+      completed: !todo.completed,
+    };
+
+    updateTodoOnServer(updatedTodo)
+      .then(todoFromServer => updateTodo(todoFromServer))
+      .catch(() => setErrorMessage(TodoError.Update))
+      .finally(() => setIsLoading(false));
```

Comment:
You could rewrite this function simply as `updateTodo`, taking the updated todo as a parameter. It would shorten your code and allow you to update the `todo` when the title has changed.